src/core/qgsmaplayer.h

```diff
@@ -2045,6 +2063,11 @@ class CORE_EXPORT QgsMapLayer : public QObject
     //! A flag that tells us whether to use the above vars to restrict layer visibility
     bool mScaleBasedVisibility = false;
 
+    /**
+     * Stores information about server properties
+     */
+    QgsMapLayerServerProperties *mServerProperties = nullptr;
```

Comment:
Can this be a unique_ptr instead?

Reply:
@dmarteau can you take a look at this @nyalldawson comment ? thanks

Reply:
Done, changed `mServerProperties` to `unique_ptr`  since `mServerProperties` is the exclusive property of QgsMapLayer and cannot be copied or shared.